plugin/trino-opa/src/main/java/io/trino/plugin/opa/schema/OpaBatchQueryResult.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.opa.schema;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNullElse;
+
+public record OpaBatchQueryResult(@JsonProperty("decision_id") String decisionId, @NotNull List<Integer> result)
+{
+    public OpaBatchQueryResult
+    {
+        result = ImmutableList.copyOf(requireNonNullElse(result, ImmutableList.of()));
```

Comment:
Yes, this depends on whether decision logging is enabled - if it is, `decision_id` will be included in the response. It is otherwise omitted ([docs](https://www.openpolicyagent.org/docs/latest/management-decision-logs/))

OPA generates its own decision_id which can be queried by the observability APIs ([docs](https://docs.styra.com/das/observability-and-audit/decision-logs/overview)). Users may also pass in one if they so wish for things like Open Tracing (we're not doing this as of this PR) https://github.com/open-policy-agent/opa/issues/1451#issuecomment-498950686